### PR TITLE
week 2 assessment complete

### PR DESCRIPTION
**Pull request by jumba23**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ x] Coding Challenges
- [ x] Coding Concepts
- [ x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,46 +14,153 @@
...
-// Expected output: "-7 is not divisible by three"
+describe("isDivisibleByThree", () => {
+  it("the function will take in an object and check if te number inside is divisible by 3", () => {
+    const object1 = { number: 15 };
+    // Expected output: "15 is divisible by three"
+    const object2 = { number: 0 };
+    // Expected output: "0 is divisible by three"
+    const object3 = { number: -7 };
+    // Expected output: "-7 is not divisible by three"
+    expect(isDivisibleByThree({ number: 15 })).toEqual(
```
Comment (elysemontano) [inline code-challenges.test.js:38]: Since your test variables are already declared, you can pass those into your function inside your expect statement.

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,46 +14,153 @@
...
+    expect(firstVowel(vowelTester2)).toEqual(0);
+    expect(firstVowel(vowelTester3)).toEqual(2);
+  });
+});
+
+//ReferenceError: firstVowel is not defined
 
 // b) Create the function that makes the test pass.
+
+// ============== <<<<< PSEUDO CODE >>>>> ===========================
```
Comment (elysemontano) [inline code-challenges.test.js:147]: Great work in this section!  Your pseudocode is great and logic is solid!  If you want to stretch yourself, try using a higher-order function for this last problem.

File: coding-concepts.js [hunk 3]
```diff
@@ -7,62 +7,60 @@
...
 }
-const learnStudent = new Learn("George")
-// console.log(learnStudent)
+const learnStudent = new Learn("George");
+console.log(learnStudent);
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: Learn { student: 'George', cohort: 'Foxtrot', year: 2022 }
+// b) Verify and explain: at the point of declaring/assigning the class Learn, the constructor method takes in one parameter -"name", which is used to set the initial value of the field this.student. The method only declares the variable student(it's undefined at this point), while declaring and assigning variables for cohort and year . When we declare and assign our new variable learnStudent we pass in an argument "George" into our new object. Now, the only place this argument can go, as previously determined by the constructor method, is to be assigned to the variable student. At this point consol.log will print our new object with 3 variables and their assigned values.
```
Comment (elysemontano) [inline coding-concepts.js:66]: Great work in this section!!  👏

File: interview-questions.md [hunk 4]
```diff
@@ -8,50 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter is used when declaring a function and holds the value we use inside the function. Usually, it is placed inside the parentheses. An argument is used when a function is called, right after the function's name and must be in the parentheses. The easiest way to express the difference between the two would be, arguments are the names and arguments are the values passed in a function.
```
Comment (elysemontano) [inline interview-questions.md:11]: Excellent and thorough answers in this section!  Crushed it!

**Review by elysemontano — APPROVED** [review]
Outstanding work Goran!  Your pseudocode and interview answers are exceptional and your code logic is solid.  If you are looking to stretch yourself, once you have your tests passing, see if you can refactor your code and or consider edge cases and writing tests for them.
